sdk/inc/azure/iot/az_iot_pnp_client.h

```diff
@@ -685,41 +685,71 @@ AZ_NODISCARD az_result az_iot_pnp_client_property_get_property_version(
     int32_t* out_version);
 
 /**
- * @brief Iteratively Read the IoT Plug and Play component properties.
+ * @brief Iteratively read the IoT Plug and Play component properties.
  *
  * Note that between calls, the #az_span pointed to by \p out_component_name shall not be modified,
  * only checked and compared. Internally, the #az_span is only changed if the component name changes
  * in the JSON document and is not necessarily set every invocation of the function.
  *
- * The `out_property_name_and_value` is passed back as an #az_json_reader. On success, the reader
- * will be pointing to the property name. After checking the property name, the reader can be
- * advanced to the property value by calling az_json_reader_next_token().
+ * On success, the `ref_json_reader` will be set on a valid property name. After checking the
+ * property name, the reader can be advanced to the property value by calling
+ * az_json_reader_next_token(). Note that on the subsequent call to this API, it is expected that
+ * the json reader will be placed AFTER the read property name and value. That means that after
+ * reading the property value (including single values or complex objects), the user must call
+ * az_json_reader_next_token().
+ *
+ * Below is a code snippet which you can use as a starting point:
+ *
+ * @code
+ *
+ * while (az_result_succeeded(az_iot_pnp_client_property_get_next_component_property(
+ *       &pnp_client, &jr, response_type, &component_name)))
+ * {
+ *   // Check if property is of interest (sub user_property for your own)
+ *   if (az_json_token_is_text_equal(
+ *           &jr.token, user_property))
+ *   {
+ *     az_json_reader_next_token(&jr);
+ *
+ *     // Get the property value here
+ *
+ *     // Skip to next property value
+ *     az_json_reader_next_token(&jr);
+ *   }
+ *   else
+ *   {
+ *     // The JSON reader must be advanced regardless of whether the property
+ *     // is of interest or not.
+ *     // Skip children in case the property value is an object
+ *     az_json_reader_skip_children(&jr);
+ *     az_json_reader_next_token(&jr);
```

Comment:
this should be :
next_token;  // for property name
next_token/skip_children; // for value

Reply:
This stems from this:
https://github.com/danewalton/azure-sdk-for-c/pull/10#discussion_r517729633

I can add the skip here too to be explicit.